File: mtc2umati/mtc2umati/ConfigStore.cs
using System;
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;


namespace mtc2umati
{
    public static class ConfigStore
    {
        public static VendorConfig VendorSettings { get; set; } = new VendorConfig();

        public static void LoadConfigJSON(string vendor)
        {
            string configPath = "./config.json";
            if (!File.Exists(configPath))
            {
                throw new FileNotFoundException($"Configuration file not found at {configPath}");
            }

            string json = File.ReadAllText(configPath);

            var config = JsonConvert.DeserializeObject<Dictionary<string, Dictionary<string, string>>>(json);

            if (config == null || !config.ContainsKey(vendor))
            {
                throw new KeyNotFoundException($"Vendor '{vendor}' not found in config.");
            }

            VendorSettings = new VendorConfig
            {
                MTCServerIP = config[vendor]["MTConnectServerIP"],
                MTCServerPort = int.Parse(config[vendor]["MTConnectServerPort"]),
                MTCNamespace = config[vendor]["MTConnectNamespace"],
                Mapping_file = config[vendor]["Mapping_file"],
                Mapping_sheet = config[vendor]["Mapping_sheet"],
                Information_model = config[vendor]["Information_model"],
                OPCNamespace = config[vendor]["OPCNamespace"],
                Machine_Name = config[vendor]["Machine_Name"],
                Mode = int.Parse(config[vendor]["Mode"])
            };
        }
    }

    public class VendorConfig
    {
        public string? MTCServerIP { get; set; }
        public int MTCServerPort { get; set; }
        public string? MTCNamespace { get; set; }
        public string? Mapping_file { get; set; }
        public string? Mapping_sheet { get; set; }
        public string? Information_model { get; set; }
        public string? OPCNamespace { get; set; }
        public string? Machine_Name { get; set; }
        public int Mode { get; set; }
    }
}

File: umati2mtc/helper/config_store.py
# ========================================================================
# Copyright (c) 2025 Aleks Arzer, Institut für Fertigungstechnik und Werkzeugmaschinen, Leibniz Universität Hannover
# ========================================================================

import json
import os


class ConfigStore:
    MTConnectServerIP = None
    MTConnectServerPort = 0
    MQTTServerIP = None
    MQTTPort = 0
    Mapping_file = None
    Mapping_sheet = None
    Information_model = None
    DevicestreamName = None
    MTConnectNamespace = None
    Gateway_topic_prefix = None

    @staticmethod
    def load_config_json(vendor: str):
        config_path = "./config.json"
        if not os.path.exists(config_path):
            raise FileNotFoundError(
                f"[ERROR] Configuration file not found at {config_path}"
            )

        with open(config_path, "r", encoding="utf-8") as file:
            config = json.load(file)

        if vendor not in config:
            raise KeyError(f"[ERROR] Vendor '{vendor}' not found in config.")

        value = config[vendor]

        ConfigStore.MTConnectServerIP = value["MTConnectServerIP"]
        ConfigStore.MTConnectServerPort = int(value["MTConnectServerPort"])
        ConfigStore.MQTTServerIP = value["MQTTServerIP"]
        ConfigStore.MQTTPort = int(value["MQTTPort"])
        ConfigStore.Mapping_file = value["Mapping_file"]
        ConfigStore.Mapping_sheet = value["Mapping_sheet"]
        ConfigStore.Information_model = value["Information_model"]
        ConfigStore.DevicestreamName = value["DevicestreamName"]
        ConfigStore.MTConnectNamespace = value["MTConnectNamespace"]
        ConfigStore.Gateway_topic_prefix = value["Gateway_topic_prefix"]
        print(f"[INFO] Configuration loaded for vendor '{vendor}'")
